### ISSUE TITLE
Training stopping after 6th epoch

### ISSUE THREAD

**Issue by rohanbanerjee** [Jianf-Wang/GBDL#4]
Hello authors, thank you for your interesting work and providing a well-documented code along with it. I am trying to reproduce to results in the paper. For this, as a starting point, I'm following the steps mentioned in the README. I have downloaded the data from the provided drive link.

Though the training starts without any issues, it fails to at the 6th epoch with the following traceback. Would you have any pointers? Any help is appreciated, happy to provide more details.

![Screen Shot 2023-04-01 at 3 01 31 PM](https://user-images.githubusercontent.com/25586344/229616901-dd203819-ba6a-41c1-95cb-a7ebc5f074e3.png)

Thanks,
Rohan Banerjee

**Comment by Jianf-Wang** [comment]
Hi. In fact, I did not meet this issue before.. Currently, I do not have any server to debug the code, and I am not quite sure whether I can help.  What is your batch size ? How many gpus are used for training ?

**Comment by rohanbanerjee** [comment]
Thanks for your reply! I am using the default values, batch_size = 4 and number of GPUs = 1 (2 num_workers). This is what I am using for running the training script

`python train.py  --name bayes_AtriaSeg_16 --train_txt ./train_AtriaSeg.txt --val_txt ./test_AtriaSeg.txt --label_factor_semi 0.5 --epochs 10`

**Comment by Jianf-Wang** [comment]
Would that happen if you use more than 1 gpu for training ? (such as 2 or 4).

**Comment by rohanbanerjee** [comment]
So I started the training on 4 GPUs (P100) and also changed the label_factor_semi to 0.1 -- which is basically the default value. The issue still persists.

**Comment by Jianf-Wang** [comment]
![result](https://user-images.githubusercontent.com/32588665/229676779-a67e2636-7a4b-47c0-b8dc-b21542637423.PNG)
Hi, I just borrowed some gpus from another server. I did not get your errors. Everything works fine. It is very strange ... How about your enviornment ? This link could be helpful : https://github.com/pytorch/pytorch/issues/31460

**Comment by rohanbanerjee** [comment]
Thanks so much, Jianfeng. I have the training running now. The issue was with the batch size, the default batch size of 4 was giving the error, I changed it to 7 which was a silly mistake. Setting it as 8 solved the issue!

**Comment by Jianf-Wang** [comment]
is there any issue for training on a single GPU ? I did not test it before

**Comment by rohanbanerjee** [comment]
Hi Jianfeng, I tried running this on a single GPU with a batch size of 4 (default), it shows out of memory. In fact, when I was using all my 8 GPUs for training with a batch size of 4, it was giving me the above stated error, it only worked when I changed my batch size to 8, which is a little strange. I see that you have also used `drop_last` which should technically not give the mentioned error whatsoever in my understanding.

Sorry for the late reply!
